bust-a-move-framework/src/main/java/bustamove/screen/OptionScreen.java

```diff
@@ -0,0 +1,98 @@
+package bustamove.screen;
+
+import bustamove.screen.attributes.Button;
+import bustamove.screen.config.GameConfig;
+import bustamove.screen.config.GameState;
+import bustamove.system.Log;
+import bustamove.system.SoundHandler;
+import org.newdawn.slick.GameContainer;
+import org.newdawn.slick.SlickException;
+import org.newdawn.slick.state.StateBasedGame;
+
+/**
+ * The OptionScreen class display the option menu.
+ * @author Winer bao
+ */
+public class OptionScreen extends Screen {
+    /**
+     * Getter method: for the GameState ID.
+     * @return integer of BasicGameState number.
+     */
+    public final int getID() {
+        return GameState.OPTION_SCREEN;
+    }
+
+    /**
+     * Initializes a option screen object.
+     */
+    public OptionScreen() {
+        Log.getInstance().log(this, "Option Screen initialised");
+    }
+
+    /**
+     * Called when BasicGameState initializes.
+     * @param game the game container
+     * @param stateBasedGame the state based game
+     * @throws SlickException any type of slick exception
+     */
+    public final void init(final GameContainer game,
+                           final StateBasedGame stateBasedGame)
+            throws SlickException {
+        initBasicScreen(game);
+        Button mute = getMuteUnmuteButton();
+        getButtons().add(mute);
+        Button back = new Button("Back", GameConfig.FIFTH_LINE,
+                GameConfig.WIDTH2, GameConfig.HEIGHT);
+        back.addGameStateChangeAction(stateBasedGame,
+                GameState.MAIN_MENU);
+        getButtons().add(back);
+        super.center();
+    }
+
+    /**
+     * Updates the text of the mute button and calls update() of super
+     * class.
+     * @param game the game container
+     * @param stateBasedGame the state based game
+     * @param i delta of time exceeded
+     * @throws SlickException any Slick exceptions
+     */
+    public final void update(final GameContainer game,
+                             final StateBasedGame stateBasedGame,
+                             final int i)
+            throws SlickException {
+        super.updateBasicScreen(game, stateBasedGame, i);
+        getButtons().set(0, getMuteUnmuteButton());
+    }
+
+    /**
+     * Create a mute or unmute button.
+     * @return Button
+     */
+    private Button getMuteUnmuteButton() {
```

Comment:
A nicer approach would be if you used 1 button. The isMuted boolean you check below here could move inside the run(). And if you made the button final you are able to use button functions like setText inside of the run() function. That would also fix the jumping of the button.